app/src/main/java/org/mozilla/FirefoxLite/firefoxdownloader/SampleService.java

```diff
@@ -0,0 +1,249 @@
+        Log.d("nnull","unn");
+        String downloadstatus = intent.getStringExtra("downloadstatus");
+        String schedulestatus = intent.getStringExtra("schedulestatus");
+        String removeschedulling = intent.getStringExtra("scheduleoption");
+        if(removeschedulling!=null){
+            if(removeschedulling.equals("removeschedulling")){
+                MozzDownloader.cancel(downloadIdOne);
+                Notification notification=new NotificationCompat.Builder(getApplicationContext(),channel1ID)
+                        .setContentTitle("mandy.pmd").setContentText("Schedule Terminated").setSmallIcon(R.drawable.ic_launcher_background).build();
+                startForeground(1,notification);
+                return START_STICKY;
+            }
+        }
+        if(schedulestatus!=null)
+        {
+            if(schedulestatus.equals("scheduled")){
+
+                Notification notification=new NotificationCompat.Builder(getApplicationContext(),channel1ID)
+                        .setContentTitle("mandy.pmd").setContentText("Download Schedulled").setSmallIcon(R.drawable.ic_launcher_background).build();
+                startForeground(1,notification);
+                if(downloadstatus!=null){
+                    if(downloadstatus.equals("cancelled")){
+                        MozzDownloader.cancel(downloadIdOne);
+                        return START_STICKY;
+                    }
+                    if (Status.QUEUED == MozzDownloader.getStatus(downloadIdOne)) {
+                        MozzDownloader.pause(downloadIdOne);
+                        return START_STICKY;
+                    }
+
+                }
+                if (Status.PAUSED == MozzDownloader.getStatus(downloadIdOne)) {
+                    MozzDownloader.resume(getApplicationContext(), this, downloadIdOne);
+                    return START_STICKY;
+                }
+                int totaldiffminutes = intent.getIntExtra("scheduletime",0);
+                downloadIdOne = MozzDownloader.download(URL1, dirPath, "mandys.pmd")
+                        .build()
+                        .setOnStartOrResumeListener(new OnStartOrResumeListener() {
+                            @Override
+                            public void onStartOrResume() {
+                                Notification notification=new NotificationCompat.Builder(getApplicationContext(),channel1ID)
+                                        .setContentTitle("mandy.pmd").setContentText("Download Running").setSmallIcon(R.drawable.ic_launcher_background).build();
+                                startForeground(1,notification);
+                                Intent intent = new Intent();
+                                intent.setAction(MY_ACTION);
+                                intent.putExtra("Statusofdownload","started");
+                                sendBroadcast(intent);
+                                sendOnchannel1("mandy.pmd", "Download Running");
```

Comment:
may I know what is this for?

Reply:
Intent send the download status from service to mainactivity so that ui is updated simulataneously with listeners

Reply:
send on channel is notification related to download number first